src/hotspot/share/gc/shenandoah/shenandoahMark.inline.hpp

```diff
@@ -58,9 +62,13 @@ void ShenandoahMark::dedup_string(oop obj, StringDedup::Requests* const req) {
 }
 
 template <class T, ShenandoahGenerationType GENERATION, StringDedupMode STRING_DEDUP>
-void ShenandoahMark::do_task(ShenandoahObjToScanQueue* q, T* cl, ShenandoahLiveData* live_data, StringDedup::Requests* const req, ShenandoahMarkTask* task) {
+void ShenandoahMark::do_task(ShenandoahObjToScanQueue* q, T* cl, ShenandoahLiveData* live_data, StringDedup::Requests* const req, ShenandoahMarkTask* task, uint worker_id) {
   oop obj = task->obj();
 
+  // TODO: This will push array chunks into the mark queue with no regard for
+  // generations. I don't think it will break anything, but the young generation
+  // scan might end up processing some old generation array chunks.
```

Comment:
Shouldn't this code push to the same queue from which it originally popped the last task? AFAIU, chunks can only cross generation boundaries if a worker actively steals it from another queue, and I don't know if that is even possible between generations.

Reply:
@earthling-amzn will delete this on his `TODO` clean up PR.